### PR TITLE
Load genesis txns

### PR DESCRIPTION
**Pull request by syuan100**
This PR is to load the genesis block txns into the db so that blockchain-node can query them. Currently the `transaction_get` for txn hash `RNaJpIU3K5NqUiQuDvgdTMQAdG9UQ0RtwkySWq-rLVE` is not available in blockchain-node. 

This code path currently triggers every time a txn hash isn't found when using `transaction_get`. It wasn't clear if there was another place to modify the code so that the genesis block is loaded on init, or if that might be a blockchain-core change. Open to suggestions to modify.

Required for proper Rosetta API configuration here: https://github.com/helium/rosetta-helium/pull/22

### CODE REVIEW

**Comment by abhay** [comment]
Maybe something like this could work instead?

```diff
diff --git a/src/bn_txns.erl b/src/bn_txns.erl
index f01efac..cb03be5 100644
--- a/src/bn_txns.erl
+++ b/src/bn_txns.erl
@@ -54,8 +54,22 @@ follower_height(#state{db = DB, default = DefaultCF}) ->
         {error, _} = Error -> ?jsonrpc_error(Error)
     end.
 
-load_chain(_Chain, State = #state{}) ->
-    {ok, State}.
+load_chain(_Chain, State0 = #state{}) ->
+    maybe_load_genesis(State0).
+
+maybe_load_genesis(State0 = #state{}) ->
+    case blockchain:get_block(1, blockchain_worker:blockchain()) of
+        {ok, Block} ->
+            Hash = blockchain_txn:hash(lists:last(blockchain_block:transactions(Block))),
+            case get_transaction(Hash, State0) of
+                {ok, _} -> % already loaded
+                    {ok, State0};
+                _ -> % attempt to load
+                    load_block([], Block, [], [], State0)
+            end;
+        Error ->
+            Error
+    end.
```

note: did not test out the edge cases.

**Review by abhay — COMMENTED** [review]
Probably don't need the `State0` aliases I did. `State` would probably do it. Does it work for the needs?

**Comment by syuan100** [comment]
Yes it does

**Comment by syuan100** [comment]
Ok made just got around to making the update. tested and works

**Comment by abhay** [comment]
Seems reasonable. @madninja thoughts?

**Comment by abhay** [comment]
@madninja you had some concerns about this approach. Could you mention them here?

**Comment by madninja** [comment]
> @madninja you had some concerns about this approach. Could you mention them here?

Yeah, I don't get why genesis isn't loaded at all to begin with? Most followers (including ETL) get block 1 on startup. Why are you not seeing this?

File: src/bn_txns.erl [hunk 1]
```diff
@@ -55,7 +55,21 @@ follower_height(#state{db = DB, default = DefaultCF}) ->
     end.
 
 load_chain(_Chain, State = #state{}) ->
-    {ok, State}.
+    maybe_load_genesis(State).
+
+maybe_load_genesis(State = #state{}) ->
+    case blockchain:get_block(1, blockchain_worker:blockchain()) of
```
Comment (madninja) [inline src/bn_txns.erl]: why not use the Chain that is passed in `load_chain` higher up?

Comment (abhay) [reply src/bn_txns.erl]: Ah yes. Good catch.

File: src/bn_txns.erl [hunk 2]
```diff
@@ -55,7 +55,21 @@ follower_height(#state{db = DB, default = DefaultCF}) ->
     end.
 
 load_chain(_Chain, State = #state{}) ->
-    {ok, State}.
+    maybe_load_genesis(State).
```
Comment (abhay) [inline src/bn_txns.erl]:
```suggestion
    maybe_load_genesis(Chain, State).
```

File: src/bn_txns.erl [hunk 3]
```diff
@@ -55,7 +55,21 @@ follower_height(#state{db = DB, default = DefaultCF}) ->
     end.
 
 load_chain(_Chain, State = #state{}) ->
```
Comment (abhay) [inline src/bn_txns.erl]:
```suggestion
load_chain(Chain, State = #state{}) ->
```

File: src/bn_txns.erl [hunk 4]
```diff
@@ -55,7 +55,21 @@ follower_height(#state{db = DB, default = DefaultCF}) ->
...
-    {ok, State}.
+    maybe_load_genesis(State).
+
+maybe_load_genesis(State = #state{}) ->
```
Comment (abhay) [inline src/bn_txns.erl]:
```suggestion
maybe_load_genesis(Chain, State = #state{}) ->
```

File: src/bn_txns.erl (hunk 1 above)
Comment (abhay) [inline src/bn_txns.erl]:
```suggestion
    case blockchain:get_block(1, Chain) of
```

**Review by abhay — CHANGES_REQUESTED** [review]
This should do it, right, @madninja ?

**Review by abhay — APPROVED** [review]

**Comment by abhay** [comment]
Will you squash these commits and rebase the branch, @syuan100?
